tests/integration_test.rs

```diff
@@ -33,6 +36,195 @@ macro_rules! assert_matches {
     };
 }
 
+#[tokio::test]
+async fn phoenix_channels_socket_status_test() {
+    let _ = env_logger::builder()
+        .parse_default_env()
+        .filter_level(log::LevelFilter::Debug)
+        .is_test(true)
+        .try_init();
+
+    let id = id();
+    let url = shared_secret_url(id);
+    let socket = Socket::spawn(url).await.unwrap();
+
+    assert_eq!(socket.status(), socket::Status::NeverConnected);
+    assert_eq!(socket.has_never_connected(), true);
+
+    socket.connect(CONNECT_TIMEOUT).await.unwrap();
+    assert_eq!(socket.status(), socket::Status::Connected);
+    assert_eq!(socket.is_connected(), true);
+
+    socket.disconnect().await.unwrap();
+    assert_eq!(socket.status(), socket::Status::Disconnected);
+    assert_eq!(socket.is_disconnected(), true);
+
+    socket.shutdown().await.unwrap();
+    assert_eq!(socket.status(), socket::Status::ShutDown);
+    assert_eq!(socket.is_shutdown(), true);
+}
+
+#[tokio::test]
+async fn phoenix_channels_socket_event_test() {
+    let _ = env_logger::builder()
+        .parse_default_env()
+        .filter_level(log::LevelFilter::Debug)
+        .is_test(true)
+        .try_init();
+
+    let id = id();
+    let url = shared_secret_url(id);
+    let socket = Socket::spawn(url).await.unwrap();
```

Comment:
This can use the ? syntax if returning a Result<(), Error> with an error union made by thiserror

Reply:
How do I get `thiserror` to make that union?

Reply:
Add each error type into an enum while using the thiserror::Error derive and with the #[from] annotation on each error type. It would also work to use thiserror

Reply:
Anyhow might be easier for this as there are a lot of cases

Reply:
```
#[derive(thiserror::Error, Debug)]
enum Error {
    #[error(transparent)]
    Elapsed(#[from] Elapsed),
    #[error(transparent)]
    JoinError(#[from] JoinError),
    #[error(transparent)]
    ShutdownError(#[from] ShutdownError),
    #[error(transparent)]
    SocketError(#[from] SocketError),
    #[error(transparent)]
    Tungstenite(#[from] tungstenite::Error),
}
```

Reply:
I'll open a separate issue to cover this change to these and the pre-existing tests.